File: CodingTestPlatform/BOJ/2_Silver/4/10845/10845.py
# 큐
# https://www.acmicpc.net/problem/10845

from collections import deque
import sys
input = sys.stdin.readline


class Queue:
    def __init__(self):
        self.q = deque([])

    def push(self, v):
        self.q.append(v)

    def pop(self):
        if len(self.q) == 0:
            return -1
        else:
            return self.q.popleft()

    def size(self):
        return len(self.q)

    def empty(self):
        if len(self.q) == 0:
            return 1
        else:
            return 0

    def front(self):
        if len(self.q) == 0:
            return -1
        else:
            return self.q[0]

    def back(self):
        if len(self.q) == 0:
            return -1
        else:
            return self.q[len(self.q) - 1]


N = int(input().rstrip())
q = Queue()
for i in range(N):
    c = input().rstrip().split()
    if c[0] == 'push':
        q.push(c[1])
    elif c[0] == 'pop':
        print(q.pop())
    elif c[0] == 'size':
        print(q.size())
    elif c[0] == 'empty':
        print(q.empty())
    elif c[0] == 'front':
        print(q.front())
    elif c[0] == 'back':
        print(q.back())

File: CodingTestPlatform/BOJ/2_Silver/4/10845/10845.js
/**
 * 큐
 * Silver4
 * https://www.acmicpc.net/problem/10845
 */

class Node {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

class Queue {
    constructor() {
        this._head = null;
        this._tail = null;
        this._size = 0;
    }

    push(val) {
        const node = new Node(val);
        if (this._head === null) {
            this._head = node;
            this._tail = node;
        } else {
            this._tail.next = node;
            this._tail = node;
        }

        this._size++;
        return null;
    }

    pop() {
        if (this._head === null) return -1;

        const temp = this._head;
        this._head = this._head.next;
        if (this._head === null) {
            this._tail = null;
        }
        
        this._size--;
        return temp.val;
    }

    size() {
        return this._size;
    }

    empty() {
        return this._size === 0 ? 1 : 0;
    }

    front() {
        return this._head ? this._head.val : -1;
    }

    back() {
        return this._tail ? this._tail.val : -1;
    }
}

const [N, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split("\n");
const q = new Queue(), result = [];

input.forEach((v) => {
    const [command, val] = v.split(" ");
    const num = (val !== undefined) ? q[command](val) : q[command]();
    if (num !== null) result.push(num);
});

console.log(result.join("\n"));